server/src/main/java/com/thoughtworks/go/server/messaging/activemq/JMSMessageListenerAdapter.java

```diff
@@ -33,13 +34,14 @@
     private final MessageConsumer consumer;
     private final GoMessageListener listener;
     private final DaemonThreadStatsCollector daemonThreadStatsCollector;
+    private SystemEnvironment systemEnvironment;
     public Thread thread;
 
-    private JMSMessageListenerAdapter(MessageConsumer consumer, GoMessageListener listener, DaemonThreadStatsCollector daemonThreadStatsCollector) {
+    private JMSMessageListenerAdapter(MessageConsumer consumer, GoMessageListener listener, DaemonThreadStatsCollector daemonThreadStatsCollector, SystemEnvironment systemEnvironment) {
         this.consumer = consumer;
         this.listener = listener;
         this.daemonThreadStatsCollector = daemonThreadStatsCollector;
-
+        this.systemEnvironment = systemEnvironment;
 
         thread = new Thread(this);
```

Comment:
Since we're just spawning a background thread. I wonder if we need to handle situations where the thread dies (I believe I've seen this situation a few times, thread going OOM)

Reply:
@ketan Yes. Would be nice. But, I'll leave that to one of you. It probably needs to be managed in one place, and across other important threads too.

Reply:
Also, trying to handle OOMs is not easy. It's possible that further attempts at bringing up the thread will again cause OOMs. @maheshp suggested adding a server health message to recommend restarting GoCD.

Reply:
@maheshp I've added the server health message. I've verified that there's only one shown even if multiple queues have died.

<img width="1057" alt="2019_02_05_18-18-02" src="https://user-images.githubusercontent.com/172235/52313557-e5d2ed00-297c-11e9-8793-e8dcd62cbf4c.png">

I can change the message, if needed. I don't think most users will see it. So, it should be ok.